framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3094,15 +3095,35 @@ protected void checkThrownExpression(ThrowTree tree) {
         AnnotatedTypeMirror throwType = atypeFactory.getAnnotatedType(tree.getExpression());
         TypeMirror throwTM = throwType.getUnderlyingType();
         Set<? extends AnnotationMirror> required = getThrowUpperBoundAnnotations();
+        if (methodTree != null && getExceptionList(methodTree) != null) {
+            List<AnnotatedTypeMirror> exceptionList = getExceptionList(methodTree);
```

Comment:
This calls `getExceptionList` twice. The first call might create a new List, which is just checked for nullness and thrown away. Then the second call allocates a new list which is then used.
This is not good for memory efficiency.

Reply:
> This calls `getExceptionList` twice. The first call might create a new List, which is just checked for nullness and thrown away. Then the second call allocates a new list which is then used. This is not good for memory efficiency.

Thanks. I will add a local variable for it.